sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -403,6 +403,8 @@ case class RangePartitioning(ordering: Seq[SortOrder], numPartitions: Int)
   override def nullable: Boolean = false
   override def dataType: DataType = IntegerType
 
+  var needClusteredDistribution = false
```

Comment:
Here `needClusteredDistribution` maybe lost if the RangePartitioning is copied after the EnsureRequirement...